backend/baseclass.pm

```diff
@@ -1211,37 +1216,52 @@ sub start_ssh_serial {
     }
     $chan->blocking(0);
     $chan->pty(1);
-    $self->{select}->add($self->{serial}->sock);
+    $chan->ext_data('merge');
+    $self->{select_read}->add($ssh->sock);
     return ($ssh, $chan);
 }
 
 sub check_ssh_serial {
-    my ($self, $fh) = @_;
+    my ($self, $fh, $write) = @_;
+
+    my $ssh = $self->{serial};
+    return 0 unless $ssh;
+    my $ssh_socket = $ssh->sock;
+    return 0 unless $ssh_socket == $fh;
 
-    if ($self->{serial} && $self->{serial}->sock == $fh) {
-        my $chan = $self->{serial_chan};
-        my $line = <$chan>;
-        if (defined $line) {
-            print $line;
+    if ($write) {
```

Comment:
This method fails whenever `$write` is true but doesn't actually check anything here... I gather this error should move to the callsite?

Reply:
It checks for the existence of the SSH session and whether its socket matches `$fh`. In this case it returns `0` for "it wasn't that file descriptor". In the write case I made it return `1` because `$fd` matches and the caller shouldn't check anywhere else.